docs: Revamp launcher documentation (#44)
n8n-io/task-runner-launcher: docs/setup.md

# Setup

> This launcher is intended for deployment as a sidecar container alongside an n8n main instance container, ideally both on the same host. The launcher exposes a health check endpoint at `/healthz` on port `5680` for liveness checks, and the n8n main instance does so on port `5681`. The orchestrator (e.g. k8s) can use these to monitor the health of the launcher and of the n8n main instance.

Reviewer: Sidecar always runs on the same host. Also we recommend running one for each n8n container (main, worker).

```suggestion
This launcher is intended for deployment as a sidecar container alongside an n8n instance container(s). The launcher exposes a health check endpoint at `/healthz` on port `5680` for liveness checks. The orchestrator (e.g. k8s) can use this to monitor the health of the launcher and of the n8n main instance.

The runner should expose a health check endpoint at `/healthz` on port `5681` for liveness checks. The launcher will use this to endpoint to monitor the health of the runner, and terminate it if needed.
```

2. Create a **config file** on the host at `/etc/n8n-task-runners.json` and make this file accessible to the launcher.

```json
{
  "task-runners": [
    {
      "runner-type": "javascript",
      "workdir": "/usr/local/bin",
      "command": "/usr/local/bin/node",
      "args": [
        "/usr/local/lib/node_modules/n8n/node_modules/@n8n/task-runner/dist/start.js"
      ],
      "allowed-env": [
        "PATH",
        "N8N_RUNNERS_GRANT_TOKEN",
        "N8N_RUNNERS_N8N_URI",
        "N8N_RUNNERS_MAX_PAYLOAD",
        "N8N_RUNNERS_MAX_CONCURRENCY",
        "NODE_FUNCTION_ALLOW_BUILTIN",
        "NODE_FUNCTION_ALLOW_EXTERNAL",
        "NODE_OPTIONS"
      ]
    }
  ]
}
```

Task runner config fields:

- `runner-type` is the type of task runner, currently only `javascript` is supported.
- `workdir` is the path to directory containing the task runner binary.
- `command` is the command to execute in order to start the task runner.
- `args` are the args for the command to execute, currently the path to the task runner entrypoint.
- `allowed-env` are the env vars that the launcher is allowed to pass to the task runner.

3. Set the **environment variables** for the launcher.

> - It is required to specify an auth token by setting `N8N_RUNNERS_AUTH_TOKEN` to a secret. The launcher will use this secret to authenticate with the n8n main instance. You will also pass this `N8N_RUNNERS_AUTH_TOKEN` to the n8n main instance as well.

Reviewer: I would talk about `n8n instance` since it can be either `main` or `worker`

```suggestion
- It is required to specify an auth token by setting `N8N_RUNNERS_AUTH_TOKEN` to a secret. The launcher will use this secret to authenticate with the n8n instance. You will also pass this `N8N_RUNNERS_AUTH_TOKEN` to the n8n instance as well.
```

- Optionally, specify the launcher's log level by setting `N8N_LAUNCHER_LOG_LEVEL` to `debug`, `info`, `warn` or `error`. Default is `info`. You can also use `NO_COLOR=1` to disable color output.

- Optionally, specify the launcher's auto-shutdown timeout by setting `N8N_RUNNERS_AUTO_SHUTDOWN_TIMEOUT` to a number of seconds, or set it to `0` to disable. Default is `15`. The runner will exit after this timeout if it is idle for the specified duration, and will be re-launched on demand when the next task comes in.

> - Optionally, specify the task broker's URI (i.e. main instance's URI) by setting `N8N_TASK_BROKER_URI`. Default is `http://127.0.0.1:5679`.

Reviewer: ```suggestion
- Optionally, specify the task broker's URI (i.e. n8n instance's URI) by setting `N8N_TASK_BROKER_URI`. Default is `http://127.0.0.1:5679`.
```

> - Optionally, specify the port for the launcher's health check server by setting `N8N_LAUNCHER_HEALTCHECK_PORT`. Default is `5680`. When overriding this port, be mindful of port conflicts - by default, the n8n main instance uses `5678` for its regular server and `5679` for its task runner server, and the runner uses `5681` for its healthcheck server. 

Reviewer: ```suggestion
- Optionally, specify the port for the launcher's health check server by setting `N8N_LAUNCHER_HEALTCHECK_PORT`. Default is `5680`. When overriding this port, be mindful of port conflicts - by default, the n8n main instance uses `5678` for its regular server and `5679` for its task broker server, and the runner uses `5681` for its healthcheck server. 
```
